File: testdata/clean/safe_app.py
# Ordinary business logic. None of these are dangerous code execution — the
# identifiers merely contain "eval"/"exec" as a substring, and TRIAGE-001 must
# not flag them.


def retrieval(records, key):
    return [r for r in records if r.get(key)]


def medieval_total(items):
    return sum(item.price for item in items)


def execute_plan(steps):
    # A function whose name starts with "exec" is not the builtin interpreter.
    return [step.run() for step in steps]


def evaluate_score(values):
    return sum(values) / len(values) if values else 0

File: testdata/clean/safe_app.js
// Ordinary business logic. The identifiers contain "eval"/"exec" as a
// substring, but none is dangerous code execution — TRIAGE-001 must not flag
// them. No request access, crypto, or security TODOs here either.

function retrieval(records, key) {
  return records.filter((r) => r[key]);
}

function medievalTotal(items) {
  return items.reduce((sum, item) => sum + item.price, 0);
}

const planExecutor = {
  // A method whose name starts with "exec" is not a dynamic interpreter.
  execute(steps) {
    return steps.map((step) => step.run());
  },
};

function evaluateScore(values) {
  return values.length ? values.reduce((a, b) => a + b, 0) / values.length : 0;
}

module.exports = { retrieval, medievalTotal, planExecutor, evaluateScore };
